src/statistics/timeSpent.ts

```diff
@@ -0,0 +1,67 @@
+import { store } from "openrct2-flexui";
+import { Statistic } from "../objects/Statistic";
+
+const STATISTIC_KEY = "timePassed";
+const STATISTIC_TITLE = "Time Spent in";
+
+type TimeSpendStat = number;
+
+const timeSpentStore = store<TimeSpendStat>(0);
+
+const subscribeToTimePassing = () => {
+  context.setInterval(() => {
+    // because of the ways that Stores work,
+    // passing in the same value will not trigger a change event
+    // so we need to pass in a new value each time, and not just 1 repeatedly
+    // we're actually going to ignore the value in that store and just rely on it updating
+
+    // console.log(`one second passed`);
+    timeSpentStore.set(timeSpentStore.get() + 1);
+  }, 1000);
+};
+
+// each time the hook is called, we want to add 1 to the existing time value
+// the function expects that we'll do something with the new value and the existing value
+// but in this case, we're just going to ignore the new value and just add 1 to the existing value
+function accumulateSeconds(_newVal: TimeSpendStat, existingVal: TimeSpendStat) {
+  return existingVal + 1;
+}
+
+export const timeSpentStatistic = () => {
+  const key = STATISTIC_KEY;
+  const title = STATISTIC_TITLE;
+  const statStore = timeSpentStore;
+
+  subscribeToTimePassing();
+
+  return new Statistic({
+    key,
+    title,
+    statStore,
+    resetValue: 0,
+    formatDisplay,
+    accumulator: accumulateSeconds,
+  });
```

Comment:
Given the changes I proposed for `Statistic`, the exact code may need to change, but I will still propose legacy support for time tracking, since the keys are different.

```suggestion
  const statistic = new Statistic({
    key,
    title,
    statStore,
    resetValue: 0,
    formatDisplay,
    accumulator: accumulateSeconds,
  });
  
  // LEGACY SUPPORT FOR TRACKING TIME IN-GAME
  const legacyGameTimeKey = "openrct2-statistics.time.gameTime";
  if (context.sharedStorage.has(legacyGameTimeKey)) {
    // To prevent a conflict when importing the shared storage, wait a second beforehand
    context.setTimeout(() => {
      let legacyTime = context.sharedStorage.get(legacyGameTimeKey) as number;
      if (legacyTime >= 0) {
        statistic.gameStatStore.set(legacyTime + 1);

        // Prevent legacy value from being imported more than once
        context.sharedStorage.set(legacyGameTimeKey, -1);
      }
    }, 1000);
  }

  // LEGACY SUPPORT FOR TRACKING TIME IN PARKS
  context.subscribe("map.changed", () => {
    const legacyParkTimeKey = "openrct2-statistics.time.parkTime";
    if (context.mode == "normal" && context.getParkStorage().has(legacyParkTimeKey)) {
      // To prevent a conflict when importing the park storage, wait a second beforehand
      context.setTimeout(() => {
      let legacyTime = context.getParkStorage().get(legacyParkTimeKey) as number;
        if (legacyTime >= 0) {
          statistic.parkStatStore.set(legacyTime + 1);

          // Prevent legacy value from being imported more than once
          context.getParkStorage().set(legacyParkTimeKey, -1);
        }
      }, 1000);
    }
  });

  return statistic;
```

Reply:
in line 56 and 72, this sets the -1 - wouldn't it be better to set to `undefined` so that after the initial load it'll never find the key again?

Reply:
To be honest, I was under the assumption that after you set a key, you wouldn't be able to remove it (since there's no method for that, just to `set`). If that's the case, then yeah, `undefined` would make sense, and there wouldn't need to be checks for `legacyTime >= 0` anymore.